src/client/content_cao.cpp

```diff
@@ -989,11 +989,12 @@ void GenericCAO::step(float dtime, ClientEnvironment *env)
 			updateTextures(m_previous_texture_modifier);
 		}
 	}
-	if (!getParent() && std::fabs(m_prop.automatic_rotate) > 0.001) {
-		m_rotation.Y += dtime * m_prop.automatic_rotate * 180 / M_PI;
-		rot_translator.val_current = m_rotation;
-		updateNodePos();
-	}
+
+	// This is the child node's rotation. It is only used for automatic_rotate.
+	v3f local_rot = node->getRotation();
```

Comment:
When `getParent() == nullptr`, this code can be skipped because `local_rot.Y` keeps its old value.

Reply:
Shall I move lines 994-997 into an `if` block?

Reply:
Yes, please.

Reply:
Done